core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -1228,11 +1228,6 @@ LogicalTableScan(table=[[scott, EMP]])
     </Resource>
   </TestCase>
   <TestCase name="testAntiJoinProjectTranspose">
-    <Resource name="sql">
```

Comment:
Is this SQL incorrect? Would it be better to add "rel" instead, for example:
```
<Resource name="rel">
 <![CDATA[
RelString.....
]]>
    </Resource>
```

Reply:
Not sure if it is correct, it may be correct. But there is no direct connection between sql and plan, it is easy to make mistakes. I agree with your suggestion very much, using `rel` resource may be a great idea.

Reply:
I checked and `planBefore` should be the `rel` we want. Does that mean we don't need to add a `rel` resource?

Reply:
Yes. This SQL improves readability, but it does not perform validation, which is a risk.

Reply:
@NobiGo Agree, we can wait for others' opinions. But there's an issue here: if the expected plan can be generated using SQL, should RelFn not be used? If RelFn is used, can it still be mapped to valid SQL? However, I'm unsure whether such a restriction should be imposed on developers.